src/net/sourceforge/kolmafia/MonsterData.java

```diff
@@ -1676,13 +1629,22 @@ public boolean isDrippy() {
   }
 
   public EnumSet<EncounterType> getType() {
-    // Only the first 10 Snowmen are free
-    if (this.name.equals("X-32-F Combat Training Snowman")
-        && Preferences.getInteger("_snojoFreeFights") < 10) {
-      EnumSet<EncounterType> temp = this.type.clone();
+    // Some monsters are conditionally free
+    var forceFree =
+        switch (this.name) {
+          case "X-32-F Combat Training Snowman" -> Preferences.getInteger("_snojoFreeFights") < 10;
+          case "biker", "\"plain\" girl", "jock", "party girl", "burnout" -> Preferences.getInteger(
+                  "_neverendingPartyFreeTurns")
+              < 10;
+          default -> false;
```

Comment:
How about _speakeasyFreeFights?

Reply:
The monsters themselves aren't inherently free - you could fax or wish them and they'd cost adventures. NEP monsters are free by definition - in a standard run right now you can even reminisce one for a free scaling fight!

The snowman is actually more questionable but is essentially true since it can't be copied.